pyfr/integrators/dual/pseudo/pseudocontrollers.py

```diff
@@ -59,6 +59,15 @@ def _resid(self, rcurr, rold, dt_fac):
     def _update_pseudostepinfo(self, niters, resid):
         self.pseudostepinfo.append((self.ntotiters, niters, resid))
 
+    @property
+    def dt(self):
+        return self._dt
+
+    @dt.setter
+    def dt(self, y):
+        if self.cfg.get('solver-time-integrator', 'pseudo-controller') == 'local-pi':        
```

Comment:
This is bad form.  Implement in the appropriate child class to avoid the check.  (The base class should not need to know anything about its subclasses.)

Reply:
Shifted this to `DualPIPseudoController`